src/main/java/nl/mpcjanssen/simpletask/Simpletask.kt

```diff
@@ -718,26 +718,36 @@ class Simpletask : ThemedNoActionBarActivity() {
     }
 
     private fun deleteTasks(tasks: List<TodoItem>) {
-        showConfirmationDialog(this, R.string.delete_task_message, DialogInterface.OnClickListener { dialogInterface, i ->
+        val numTasks = tasks.size
+        var title = getString(R.string.delete_task_title)
+                    .replaceFirst(Regex("%s"), numTasks.toString())
+        val delete = DialogInterface.OnClickListener { dialogInterface, i ->
             for (t in tasks) {
                 TodoList.remove(t)
             }
             TodoList.notifyChanged(Config.todoFileName, Config.eol, m_app, true)
             invalidateOptionsMenu()
-        }, R.string.delete_task_title)
-    }
+        }
 
-    private fun archiveTasks(tasksToArchive: List<TodoItem>?, areYouSureDialog: Boolean) {
+        showConfirmationDialog(this, R.string.delete_task_message, delete, title)
+    }
 
+    private fun archiveTasks() { archiveTasks(TodoList.completedTasks, false) }
+    private fun archiveTasks(tasks: List<TodoItem>, showDialog: Boolean = true) {
```

Comment:
Overloading lets us make the calls pretty (meaning, we never need to manually pass in the showDialog boolean, although that option is still available). It has to be overloaded (not just default values) because the default bool needs to be `true` when you specify the tasks to archive and `false` when you call with no args.